File: Difficulty 3.3/HIndex/HIndex.py
n = int(input())
citations = {}
for k in range(n):
    cit = int(input()) # Number of citation for that reading
    if cit in citations:
        citations[cit] += 1
    else:
        citations[cit] = 1
total = n
maximum = 0
for m in range(n + 1):
    if total < m:
        break
    else:
        if m in citations:
            total -= citations[m]
        maximum = m
print(maximum)

File: Difficulty 3.3/HIndex/HIndex.java
import java.util.Scanner;

public class HIndex{
    public static void main (String [] args) {
        Scanner sc = new Scanner(System.in);
        int[] citations = new int[1000000001];
        int n = sc.nextInt();
        for (int i = 0; i < n; i++) {
            int cit = sc.nextInt();
            citations[cit]++;
        }

        int max = 0;
        int total = n;
        for (int i = 0; i < 1000000001; i++) {
            if (total < i) {
                break;
            } else {
                total -= citations[i];
                max = i;
            }
        }

        System.out.println(max);
    }
}
